### DISCUSSION TITLE
Detecting a car/van arriving and leaving under new 0.10.0

### DISCUSSION THREAD

**Discussion by gadget-man** [blakeblackshear/frigate#2978] [Ask A Question] [answered]
Looking at the notes for 0.10.0 RC, it looks as though the new 'stationary' object detection is going to be an excellent addition for vehicle tracking.  However, I'm struggling to work out how to best utilise it in conjunction with the previous advice on avoiding stationary objects.

What I'm trying to do: create an automation in HomeAssistant that notifies me when a vehicle a) arrives and then b) leaves a particular area of our drive.  Eventually I want to link this with 'platerecognizer' to automatically open the garage door when a recognised vehicle arrives, and then close it again when it leaves.

<img width="1003" alt="Screenshot 2022-03-16 at 18 11 28" src="https://user-images.githubusercontent.com/24406912/158659019-2993cf9a-03a8-47cf-8f16-2bef8d3ca7f8.png">

I've created 2 separate Zones named driveway and Apron (the latter being the area where the car stops), and have created an automation in HA which triggers whenever the frigate/events topic publishes an event where all the following are true:
```
{{ 'Driveway' in trigger.payload_json['after']['entered_zones']}}
{{ 'Apron' in trigger.payload_json['after']['entered_zones']}}
{{ trigger.payload_json['after']['camera'] == 'Drive' }}
{{ trigger.payload_json['after']['label'] == 'car' }}
```

However, I've realised I'm getting notifications not only when the car arrives, but also when it's just sitting on the 'Apron' zone and the 'Driveway' Zone isn't being activated.

Is there a guide on how best to use the new 'stationary' option (which I understand also affects the entered_zones and current_zones - but not sure quite how!) so that I can cleanly capture a single event when the car enters the Apron, and another one when it leaves?

**Comment by NickM-27** [comment, marked as answer]
I would suggest using the `trigger.payload_json['before']` in conjunction with some length checks on the `entered_zones` to do this. For example check that before entered_zones has driveway and after entered_zones has driveway and apron would indicate that the vehicle just arrived. For leaving I would suggest checking to make sure both driveway and apron are in `before` and `current_zone` is driveway.

**Comment by gadget-man** [comment]
Many thanks - other than the fact that I seem to get multiple notifications for the same event, that seems to have worked well. I was expecting to have to use `stationary` as a filter.

**Reply by NickM-27** [reply]
Definitely plenty of variables that can be used to tune. I wrote my own automations for notifications due to this as well.
